packages/app/src/cli/services/dev/processes/dev-session.ts

```diff
@@ -60,7 +61,10 @@ export const pushUpdatesForDevSession: DevProcessFunction<DevSessionOptions> = a
   {stderr, stdout, abortSignal: signal},
   options,
 ) => {
-  const {developerPlatformClient, app} = options
+  const {developerPlatformClient} = options
+
+  // Reload the app before starting the dev session, at this point the configuration has changed (e.g. application_url)
+  const app = await reloadApp(options.app, {stderr, stdout, signal})
```

Comment:
I'm assuming this is reloading the app state since the TOML changed? I wonder how this plays out if we try to store the tunnel URL elsewhere instead of updating all the URLs 🤔 may be a post EA thing though

Reply:
yeah, I didn't want to do anything weird until we decide what to do with updating the Toml or not. This was the safest solution for now.